src/functions-and-arrays.js

```diff
@@ -1,119 +1,223 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
-
+function maxOfTwoNumbers(numeroOne, numeroXu) {
+  if (numeroOne > numeroXu) {
+    return numeroOne;
+  } else {
+    return numeroXu;
+  }
+}
 
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
-
-function findLongestWord() {}
-
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot",
+];
 
+function findLongestWord(words) {
+  let laMasLarga = "";
+  if (words === null || (words === undefined) | (words.length === 0)) {
+    return null;
+  }
+  words.forEach((word) => {
+    if (word.length > laMasLarga.length) {
+      laMasLarga = word;
+    }
+  });
+
+  return laMasLarga;
+}
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
-
+function sumNumbers(aSumar) {
+  let aSumarrrr = 0;
+  if (aSumar.length == 0) {
+    return 0;
+  }
+  aSumar.forEach((elNumero) => {
+    aSumarrrr += elNumero;
+  });
+  return aSumarrrr;
+}
 
 // Iteration #3.1 Bonus:
```

Comment:
const mixedArr = [6, 12, "miami", 1, true, "barca", "200", "lisboa", 8, 10];

function sum(mixedArr) {
  if (mixedArr.length === 0) return 0;

  let result = 0;
  for (let i = 0; i < mixedArr.length; i++) {
    const element = mixedArr[i];

    if (typeof element === "string") {
      result += element.length;
    } else if (element === true) {
      result += 1;
    } else if (typeof element === "object") {
      throw new Error("Unsupported data type sir or ma'am");
    } else {
      result += mixedArr[i];
    }
  }
  return result;
}
console.log(sum(mixedArr))